tests/integration/components/content-kit-editor/component-test.js

```diff
@@ -237,17 +240,190 @@ test('it de-links selected text and fires `on-change`', function(assert) {
+  assert.ok(this.$(`#demo-card`).length, 'Card added in display mode');
+  assert.ok(!this.$(`#demo-card-editor`).length, 'Card not in edit mode');
+  assert.ok(this.$('button#edit-card').length, 'has edit card button');
+
+  this.$('button#edit-card').click();
+  assert.ok(!this.$(`#demo-card`).length, 'Card not in display mode');
+  assert.ok(this.$(`#demo-card-editor`).length, 'Card changed to edit mode');
+});
+
+test('it has `addCardInEditMode` action to add card in edit mode', function(assert) {
+  assert.expect(2);
+  this.registry.register('template:components/demo-card',
+                         hbs`<div id="demo-card"></div>`);
+  this.registry.register('template:components/demo-card-editor',
+                         hbs`<div id="demo-card-editor"></div>`);
+  this.set('cards', [createComponentCard('demo-card')]);
+
+  this.render(hbs`
+    {{#content-kit-editor cards=cards as |contentKit|}}
+      <button id='add-card'
+              {{action contentKit.addCardInEditMode 'demo-card'}}>
+      </button>
+    {{/content-kit-editor}}
+  `);
+
+  this.$('button#add-card').click();
+
+  assert.ok(!this.$(`#demo-card`).length, 'Card added in edit mode');
+  assert.ok(this.$(`#demo-card-editor`).length, 'Card not in display mode');
+});
+
+test('`addCard` passes `data`, breaks reference to original payload', function(assert) {
+  assert.expect(6);
+
+  let passedPayload;
+
+  const DemoCardComponent = Ember.Component.extend({
+    init() {
+      this._super(...arguments);
+      passedPayload = this.get('data');
+    },
+    actions: {
+      mutatePayload() {
+        this.set('data.foo', 'baz');
+      }
+    }
+  });
+
+  this.registry.register('component:demo-card', DemoCardComponent);
+  this.registry.register('template:components/demo-card', hbs`
+    <div id="demo-card">
+      {{data.foo}}
+      <button id='mutate-payload' {{action 'mutatePayload'}}></button>
+    </div>
+  `);
+
+  this.set('cards', [createComponentCard('demo-card')]);
+  let payload = {foo: 'bar'};
+  this.set('payload', payload);
+
+  this.render(hbs`
+    {{#content-kit-editor cards=cards as |contentKit|}}
+      <button id='add-card' {{action contentKit.addCard 'demo-card' payload}}>
+      </button>
+    {{/content-kit-editor}}
+  `);
+
+  this.$('button#add-card').click();
+
+  assert.ok(!!passedPayload && passedPayload.foo === 'bar',
+            'payload is passed to card');
+  assert.ok(passedPayload !== payload,
+            'card receives data payload that is not the same object');
+
+  assert.ok(this.$('button#mutate-payload').length,
+            'has mutate-payload button');
+  assert.ok(this.$(`#demo-card:contains(${payload.foo})`).length,
+            'displays passed `data`');
+  this.$('button#mutate-payload').click();
+
+  assert.equal(passedPayload.foo, 'baz', 'mutates its payload');
+  assert.equal(payload.foo, 'bar', 'payload remains unchanged');
+});
+
+test('#activeSectionTagNames is correct', function(assert) {
```

Comment:
the `activeSectionTagNames` tests are not new; they were in the wrong file before